### PR TITLE
added an identity hash method to Object

### PR DESCRIPTION
**Pull request by thomasmulvaney**
I'm not sure if this is what you were thinking in #196 but heres an attempt. It wasn't possible to make `pixie.object.Object` extend `IObject` because it has no type and I wasn't sure if that was the best way to approach the problem, so I have just added a check to see if something satisfies `IObject` otherwise `obj_hash` is called.

### CODE REVIEW

**Comment by halgari** [comment]
A slightly cleaner route is to use the default_fn behavior of the protocol functions: 

https://github.com/pixie-lang/pixie/blob/master/pixie/vm/stdlib.py#L25

I haven't benchmarked the satisfies? yet so I don't know how well it behaves, but the JIT optimizes default fns quite well.

**Comment by thomasmulvaney** [comment]
Ok great, I've changed it to use `default_fn` option. Looks much nicer.

File: pixie/vm/stdlib.py [hunk 1]
```diff
@@ -105,8 +105,14 @@ def default_str(x):
     assert isinstance(tp, Type)
     return rt.wrap(u"<inst " + tp._name + u">")
 
+def default_hash(x):
+    tp = x.type()
```
Comment (halgari) [inline pixie/vm/stdlib.py]: Do we need this? It's going to get removed by the JIT (or even RPython) but it is extra code.

**Comment by halgari** [comment]
thanks!

**Pull request merged**